spec/cdn_source_spec.rb

```diff
@@ -49,39 +47,27 @@ def print_dir(tag)
         STDERR.puts Pathname.glob(@path.join('*')).sort.join("\n")
       end
 
-      def fulfilled_future(result)
-        Concurrent::Promises.fulfilled_future(result)
-      end
-
-      def resolved_event
-        Concurrent::Promises.resolved_event
-      end
-
-      def typhoeus_http_response_future(code, headers = {}, body = '')
-        fulfilled_future(Typhoeus::Response.new(
-                           :response_code => code,
-                           :headers => headers,
-                           :response_body => body,
-        ))
-      end
-
-      def typhoeus_non_http_response_future(code)
-        fulfilled_future(Typhoeus::Response.new(
-                           :response_code => 0,
-                           :return_code => code,
-        ))
-      end
-
       @remote_dir = fixture('mock_cdn_repo_remote')
 
       @path = fixture('spec-repos/test_cdn_repo_local')
       cleanup
       save_url('http://localhost:4321/')
 
       @source = CDNSource.new(@path)
+      @source.stubs(:sleep_async).returns(nil)
```

Comment:
@jasl is this stub needed?

Reply:
I'm not sure if I'm not stub `sleep_async`, it will actual do sleep, that will made tests running longer.

Let me do an experiment for that.

Reply:
@jasl perhaps it should be `@source.expects(:sleep_async).never`? The rest of the tests shouldn't cause sleep.

Reply:
I think we should stub it, or it will do sleep

Reply:
@jasl perhaps it should be `@source.expects(:sleep_async).never`? The rest of the tests shouldn't cause sleep.

Reply:
`raises if unexpected non-HTTP error (60.0 s)` this one

Reply:
@jasl I've realized this was a missing expectation, see my latest commit.

Reply:
oh I forget that... thanks